### DISCUSSION TITLE
Why is `reloader` deprecated?

### DISCUSSION THREAD

**Discussion by DannyBen** [sinatra/sinatra#1847] [Q&A]
I just learned on [this page](https://sinatrarb.com/contrib/reloader) that the simple and easy to use `reloader` functionality is deprecated, without any built-in or easy to use alternative.

The [alternative provided](https://github.com/jeremyevans/rack-unreloader) seems to require too many surgical changes, and I would imagine that the ability to develop without stopping and starting the server must be a high priority in any modern web development environment.

What am I missing?

**Comment by dentarg** [comment]
I guess it was broken? I'm using rerun myself (https://github.com/alexch/rerun)

**Reply by dentarg** [reply]
https://github.com/sinatra/sinatra/issues/1381 was linked from the PR that deprecated it

It was consensus from active contributors to deprecate it to lower the amount of things to maintain

**Reply by DannyBen** [reply]
This is a critical functionality. The alternatives are poorly (or rarely) maintained, more complicated to use, and slower to reload.

But it confirms my fears that Sinatra maintenance is also on the downslope. That's disappointing.

**Reply by dentarg** [reply]
It sure is. We're lucky @jkowens got access to cut releases!

**Reply by DannyBen** [reply]
> It sure is. We're lucky @jkowens got access to cut releases!

No.....I just said that hoping someone would shut me up and prove me wrong with some argument on why `reloader` is not really needed....... what have I done......
